pkg/apis/extensions/types.go

```diff
@@ -899,3 +899,45 @@ type NetworkPolicyList struct {
 
 	Items []NetworkPolicy `json:"items"`
 }
+
+// +genclient=true
+// +nonNamespaced=true
+
+// StorageClass describes the parameters for a class of storage for
+// which PersistentVolumes can be dynamically provisioned.
+//
+// StorageClasses are non-namespaced; the name of the storage class
+// according to etcd is in ObjectMeta.Name.
+type StorageClass struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	// ProvisionerType indicates the type of the provisioner.
```

Comment:
Comments should reference json-style names.

Comments for this should indicate what sort of value this expects.  For example:

```
provisioner specifies which what driver is expected to handle this StorageClass.  This is an optionally-prefixed name, like a label key. For example: "kubernetes.io/gce-pd" or "kubernetes.io/aws-ebs".
```